qutebrowser/browser/webkit/downloads.py

```diff
@@ -816,27 +836,10 @@ def get_request(self, request, *, fileobj=None, filename=None,
         if suggested_fn is None:
             suggested_fn = 'qutebrowser-download'
 
-        # We won't need a question if a filename or fileobj is already given
-        if fileobj is None and filename is None:
-            filename, q = ask_for_filename(
-                suggested_fn, self._win_id, parent=self,
-                prompt_download_directory=prompt_download_directory
-            )
-
-        if fileobj is not None or filename is not None:
-            return self.fetch_request(request,
-                                      fileobj=fileobj,
-                                      filename=filename,
-                                      suggested_filename=suggested_fn,
-                                      **kwargs)
-        q.answered.connect(
-            lambda fn: self.fetch_request(request,
-                                          filename=fn,
-                                          suggested_filename=suggested_fn,
-                                          **kwargs))
-        self._postprocess_question(q)
-        q.ask()
-        return None
+        return self.fetch_request(request,
+                                  target=target,
+                                  suggested_filename=suggested_fn,
+                                  **kwargs)
```

Comment:
I see, I've considered that. As I said, I can add the code back if people notice issues (or are just unhappy), it shouldn't be an issue (even with the open download possibility, it's just a single parameter now). I'm just happy about every cleanup I can give the download code :wink: